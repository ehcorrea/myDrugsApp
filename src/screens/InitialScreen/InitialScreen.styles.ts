import styled, { css } from 'styled-components/native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { InitialBackground } from '@/assets/images';
import { Dots, Text } from '@/components';

export const Container = styled.View`
  ${({ theme }) => css`
    background-color: ${theme.colors.mainBg};
    flex: 1;
  `}
`;

export const Header = styled.View`
  ${({ theme }) => css`
    border-bottom-left-radius: ${theme.border.radius.large}px;
    border-bottom-right-radius: ${theme.border.radius.large}px;
    flex: 2;
  `}
`;

export const Banner = styled.ImageBackground.attrs({
  source: InitialBackground,
})`
  flex: 1;
`;

export const SafeArea = styled(SafeAreaView).attrs({ edges: ['bottom'] })`
  ${({ theme }) => css`
    flex: 1;
    padding-bottom: ${theme.spacings.large}px;
  `}
`;

export const Body = styled.View`
  align-items: center;
  justify-content: flex-start;
`;

export const Title = styled(Text.Title).attrs({
  color: 'black',
  size: 'huge',
})``;

export const SecondTitleWrapper = styled.View`
  ${({ theme }) =>
    css`
      margin-top: -2%;
      padding-left: ${theme.spacings.xxlarge * 5.5}px;
    `}
`;

export const Ornament = styled(Dots)`
  ${({ theme }) =>
    css`
      margin-top: ${theme.spacings.xxlarge}px;
    `}
`;

export const Footer = styled.View`
  ${({ theme }) =>
    css`
      align-items: center;
      margin-top: ${theme.spacings.medium}px;
    `}
`;
