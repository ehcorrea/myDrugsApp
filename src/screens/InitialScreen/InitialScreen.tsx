import React from 'react';
import { useNavigation } from '@react-navigation/native';

import { Button } from '@/components';

import * as S from './InitialScreen.styles';
import { ScrollView } from 'react-native';

export function InitialScreen() {
  const { navigate } = useNavigation();

  const handleLogin = () => {
    navigate('LoginScreen');
  };

  return (
    <S.Container>
      <S.Header>
        <S.Banner
          accessibilityLabel="inicial screen banner"
          resizeMode="cover"
        />
      </S.Header>

      <S.SafeArea edges={['bottom']}>
        <ScrollView bounces={false} showsVerticalScrollIndicator={false}>
          <S.Body>
            <S.Title>Beyond The</S.Title>
            <S.SecondTitleWrapper>
              <S.Title>Journey</S.Title>
            </S.SecondTitleWrapper>
            <S.Ornament />
          </S.Body>
        </ScrollView>
        <S.Footer>
          <Button label="Get Started" rounded="large" onPress={handleLogin} />
        </S.Footer>
      </S.SafeArea>
    </S.Container>
  );
}
