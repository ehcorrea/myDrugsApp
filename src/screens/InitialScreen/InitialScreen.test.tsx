import { render } from '@/test/utils';

import { InitialScreen } from './InitialScreen';

test('should render screen with success', () => {
  const container = render(<InitialScreen />);
  expect(container.getByLabelText('inicial screen banner')).toBeTruthy();
  expect(container.getByLabelText('Get Started')).toBeTruthy();
});
