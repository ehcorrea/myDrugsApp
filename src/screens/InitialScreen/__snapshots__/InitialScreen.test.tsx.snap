// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render screen with success 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#EFEFEF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderBottomLeftRadius": 30,
          "borderBottomRightRadius": 30,
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Image
        accessibilityLabel="inicial screen banner"
        resizeMode="cover"
        source={
          {
            "testUri": "../../../src/assets/images/initialBackground.jpeg",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      [
        "bottom",
      ]
    }
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 20,
        },
      ]
    }
  >
    <RCTScrollView
      bounces={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Text
            color="black"
            size="huge"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "PlayfairDisplay-Regular",
                  "fontSize": 63,
                  "fontWeight": "600",
                },
                {},
              ]
            }
          >
            Beyond The
          </Text>
          <View
            style={
              [
                {
                  "marginTop": "-2%",
                  "paddingLeft": 176,
                },
              ]
            }
          >
            <Text
              color="black"
              size="huge"
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "PlayfairDisplay-Regular",
                    "fontSize": 63,
                    "fontWeight": "600",
                  },
                  {},
                ]
              }
            >
              Journey
            </Text>
          </View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                },
                {
                  "marginTop": 32,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#000000",
                    "borderRadius": 10,
                    "height": 8,
                    "width": 20,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#545454",
                    "borderRadius": 10,
                    "height": 8,
                    "marginLeft": 2,
                    "width": 8,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#545454",
                    "borderRadius": 10,
                    "height": 8,
                    "marginLeft": 2,
                    "width": 8,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#545454",
                    "borderRadius": 10,
                    "height": 8,
                    "marginLeft": 2,
                    "width": 8,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        [
          {
            "alignItems": "center",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Get Started"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderRadius": 30,
            "minWidth": 222,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <Text
          color="white"
          style={
            [
              {
                "color": "#FAFAFA",
                "fontFamily": "Poppins-Regular",
                "fontSize": 31,
              },
            ]
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
